packages/ava/src/data/statistics/index.ts

```diff
@@ -0,0 +1,269 @@
+/**
+ * Statistical methods used internally by ava.
+ */
+
+import { assert } from '../utils';
+
+import * as cache from './caches';
+
+/**
+ * Calculate the minimum of the array.
+ * @param value - The array to process
+ */
+export function min(value: number[]): number {
+  const cachedValue = cache.get<number>(value, 'min');
+  if (cachedValue !== undefined) {
+    return cachedValue;
+  }
+  return cache.set(value, 'min', Math.min(...value));
```

Comment:
Do we need declare constants for key?

Reply:
It doesn't matter. Function names cannot use constants, and these names are only used here.